lib/rouge/lexers/batchfile.rb

```diff
@@ -0,0 +1,163 @@
+        )
+      end
+
+      def self.devices
+        @devices ||= %w(
+          con prn aux nul com1 com2 com3 com4 com5 com6 com7 com8 com9 lpt1 lpt2
+          lpt3 lpt4 lpt5 lpt6 lpt7 lpt8 lpt9
+        )
+      end
+
+      def self.builtin_commands
+        @builtin_commands ||= %w(
+          assoc attrib break bcdedit cacls cd chcp chdir chkdsk chkntfs choice
+          cls cmd color comp compact convert copy date del dir diskpart doskey
+          dpath driverquery echo endlocal erase fc find findstr format fsutil
+          ftype gpresult graftabl help icacls label md mkdir mklink mode more
+          move openfiles path pause popd print prompt pushd rd recover ren
+          rename replace rmdir robocopy setlocal sc schtasks shift shutdown sort
+          start subst systeminfo takeown tasklist taskkill time timeout title
+          tree type ver verify vol xcopy waitfor wmic
+        )
+      end
+
+      def self.other_commands
+        @other_commands ||= %w(
+          addusers admodcmd ansicon arp at bcdboot bitsadmin browstat certreq
+          certutil change cidiag cipher cleanmgr clip cmdkey compress convertcp
+          coreinfo csccmd csvde cscript curl debug defrag delprof deltree devcon
+          diamond dirquota diruse diskshadow diskuse dism dnscmd dsacls dsadd
+          dsget dsquery dsmod dsmove dsrm dsmgmt dsregcmd edlin eventcreate
+          expand extract fdisk fltmc forfiles freedisk ftp getmac gpupdate
+          hostname ifmember inuse ipconfig kill lgpo lodctr logman logoff
+          logtime makecab mapisend mbsacli mem mountvol moveuser msg mshta
+          msiexec msinfo32 mstsc nbtstat net net1 netdom netsh netstat nlsinfo
+          nltest now nslookup ntbackup ntdsutil ntoskrnl ntrights nvspbind
+          pathping perms ping portqry powercfg pngout pnputil printbrm prncnfg
+          prnmngr procdump psexec psfile psgetsid psinfo pskill pslist
+          psloggedon psloglist pspasswd psping psservice psshutdown pssuspend
+          qbasic qgrep qprocess query quser qwinsta rasdial reg reg1 regdump
+          regedt32 regsvr32 regini reset restore rundll32 rmtshare route rpcping
+          run runas scandisk setspn setx sfc share shellrunas shortcut sigcheck
+          sleep slmgr strings subinacl sysmon telnet tftp tlist touch tracerpt
+          tracert tscon tsdiscon tskill tttracer typeperf tzutil undelete
+          unformat verifier vmconnect vssadmin w32tm wbadmin wecutil wevtutil
+          wget where whoami windiff winrm winrs wpeutil wpr wusa wuauclt wscript
+        )
+      end
+
+      def self.attributes
+        @attributes ||= %w(
+          on off disable enableextensions enabledelayedexpansion
+        )
+      end
+
+      state :basic do
+        # Comments
+        rule %r/\brem\b.*$/i, Comment
+        # Empty Labels
+        rule %r/^::.*$/, Comment
+
+        # Labels
+        rule %r/:[a-z]+/i, Name::Label
+
+        rule %r/([a-z]\w*)(\.exe|com|bat|cmd|msi)?/i do |m|
```

Comment:
@carlos-montiers I'm sorry because I should have brought this up sooner but I realised that, for performance reasons, doing all that regex interpolation isn't the best approach. This approach avoids the interpolation but should produce the same result. The results look fine to me but let me know what you think.

Reply:
Really nice. I like the new version as I learned a new way of do it with more performance. Also I updated the visual sample with code that works correctly, thus is a valid sample.